### PR TITLE
feat(error/appBoundary): make error appBoundary more resilient and prepared

### PR DESCRIPTION
**Pull request by midudev**
Please review @SUI-Components/developers 

- [x] Add tolerance error ratio in order to avoid infinite loops for errors on `render()`s methods.
- [x] Add the possibility to redirect to a different page if you have an error.
- [x] Add comments for propTypes.

### CODE REVIEW

**Review by davidbarna — APPROVED** [review]

File: components/error/appBoundary/src/index.js [hunk 1]
```diff
@@ -3,10 +3,24 @@ import PropTypes from 'prop-types'
...
-    this.props.onError({ errorMessage, errorStack })
-    // import async the alert basic component
+    const { errorThreshold, onError, redirectUrlOnBreakingThreshold } = this.props
+    const { errorCount } = this.state
+
+    onError({ errorMessage, errorStack })
+    this.setState({ errorCount: errorCount + 1 })
+
+    return errorCount >= errorThreshold
+      ? redirectUrlOnBreakingThreshold && (window.location.href = redirectUrlOnBreakingThreshold)
```
Comment (ruben-martin-lozano) [inline components/error/appBoundary/src/index.js:15]: Does the "componentDidCatch" occurs only in client side?

Comment (ruben-martin-lozano) [reply components/error/appBoundary/src/index.js:15]: If so, then it's ok, but if dont', this window.location.href would fail in server side

Comment (midudev) [reply components/error/appBoundary/src/index.js:15]: componentDidCatch doesn't catch errors for server side rendering.

https://reactjs.org/docs/error-boundaries.html#introducing-error-boundaries

Comment (ruben-martin-lozano) [reply components/error/appBoundary/src/index.js:15]: great! thx :)

**Review by ruben-martin-lozano — APPROVED** [review]

File: components/error/appBoundary/src/index.js (hunk 1 above)
Comment (midudev) [reply components/error/appBoundary/src/index.js:15]: Thanks to you!

**Pull request merged**